Wangscape/utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <type_traits>
+#include <stdexcept>
+
+namespace
+{
+
+template<typename I>
+I _pow(I base, I exp)
+{
+    I result = 1;
+    for (int i = 0; i < std::numeric_limits<I>::digits; i++)
+    {
+        if (exp & 1)
+            result *= base;
+        else if (exp == 0)
+            break;
+        exp >>= 1;
```

Comment:
Also, for the sake of clarity and compatibility with unexpected types it may be better to use named `constexpr` values instead of literal `0`, `1`, `2` in `_pow`.